File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/IStoredItem.java
package ru.kmorozov.gbd.core.config;

import java.io.File;
import java.io.IOException;
import java.io.OutputStream;

public interface IStoredItem {

    OutputStream getOutputStream() throws IOException;

    boolean exists() throws IOException;

    void delete() throws IOException;

    void close() throws IOException;

    void write(byte[] bytes, int read) throws IOException;

    File asFile();
}

File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/config/IStoredItem.kt
package ru.kmorozov.gbd.core.config

import java.io.File
import java.io.IOException
import java.io.OutputStream

interface IStoredItem {

    val outputStream: OutputStream?

    @Throws(IOException::class)
    fun exists(): Boolean

    @Throws(IOException::class)
    fun delete()

    @Throws(IOException::class)
    fun close()

    @Throws(IOException::class)
    fun write(bytes: ByteArray, len: Int)

    fun flush()

    fun asFile(): File

    fun isImage(): Boolean

    fun validate(): Boolean

    val createdNew: Boolean

    val pageNum: Int
        get() = -1
}
